lib/reactjs_jsx_ppx_v2.ml

```diff
@@ -27,36 +27,57 @@
   `ReactDOMRe.createElement(ReasonReact.fragment, [|foo|])`
 *)
 
-(*
-  This file's shared between the Reason repo and the BuckleScript repo. In
-  Reason, it's in src. In BuckleScript, it's in jscomp/bin. We periodically
-  copy this file from Reason (the source of truth) to BuckleScript, then
-  uncomment the #if #else #end cppo macros you see in the file. That's because
-  BuckleScript's on OCaml 4.02 while Reason's on 4.04; so the #if macros
-  surround the pieces of code that are different between the two compilers.
-
-  When you modify this file, please make sure you're not dragging in too many
-  things. You don't necessarily have to test the file on both Reason and
-  BuckleScript; ping @chenglou and a few others and we'll keep them synced up by
-  patching the right parts, through the power of types(tm)
-*)
-
-
+let rec find_opt p = function
+  | [] -> None
+  | x :: l -> if p x then Some x else find_opt p l
 let nolabel = ""
 let labelled str = str
+let optional str = "?" ^ str
+let isOptional str = str <> "" && String.sub str 0 1 = "?"
```

Comment:
could be simplified as `str.[0]='?'`
IIRC such optional label thing is explicitly modeled in 4.06.1, be careful when you do the upgrade

Reply:
Yup this was actually downgraded to work with 4.02. I have the original source for 4.06 using explicit labels in an old commit and will switch back and test thoroughly.